src/topology/bounded_continuous_function.lean

```diff
@@ -440,5 +440,20 @@ by rw normed_group.dist_eq; exact @norm_coe_le_norm _ _ _ _ (f-g) x
 lemma coe_le_coe_add_dist {f g : α →ᵇ ℝ} : f x ≤ g x + dist f g :=
 sub_le_iff_le_add'.1 $ (abs_le.1 $ @dist_coe_le_dist _ _ _ _ f g x).2
 
+/-- Constructing an element of a normed group from a continuous bounded function -/
+def normed_group.mk {α : Type u} {β : Type v} [topological_space α] [normed_group β]
+  (f : α  → β) (C : ℝ) (H : ∀x, norm (f x) ≤ C) (Hf : continuous f) : α →ᵇ β :=
+⟨λn, f n, ⟨Hf, ⟨C + C, λ m n,
+  calc dist (f m) (f n) ≤ dist (f m) 0 + dist (f n) 0 : dist_triangle_right _ _ _
+       ... = norm (f m) + norm (f n) : by simp
+       ... ≤ C + C : add_le_add (H m) (H n)⟩⟩⟩
```

Comment:
This name seems very confusing. The comment says it's constructing an element of a normed group, the theorem name says it's constructing a normed group, and the theorem is actually constructing a bounded continuous function. It seems like this should be in the BCF namespace, perhaps `bcf.of_normed_group`?

Reply:
This is in the bounded_continuous_function namespace, of course. In the current situation, the bounded_continuous_function is a normed_group, so this is indeed constructing a normed_group element. The difference with the other constructor is that the assumption is a pointwise bound `∀x, norm (f x) ≤ C` instead of a pair distance control `∀x y, dist (f x) (f y) ≤ C`, i.e., the assumption only makes sense in a normed group setting. Still, if you insist I should change it, tell me, I will do it.

Reply:
I can see what the point of the definition is, and I think it's a fine theorem, but the commentary and naming was confusing to me as I said. The way I would say it is that we can construct a bcf with a proof based on norms when the underlying metric space is a normed group. I think the fact that bcfs themselves form a normed group is beside the point for this theorem, although it does explain why the comment says that. I think of this as more analogous to the metric space equivalent characterization theorems. You can't join namespaces - `bcf.normed_group.T` is not related to `normed_space.T`, so I would suggest just a plain name in the bcf namespace. `of_X` is one way to indicate that it's constructing a bcf.

Reply:
OK, will change.